source/elements/oneMKL/source/domains/spblas/operations/spmv.rst

```diff
@@ -0,0 +1,378 @@
+
+       void init_spmv_descr (sycl::queue                       &queue,
+                             oneapi::mkl::sparse::spmv_descr_t *p_spmv_descr);
+
+   }
+
+.. container:: section
+
+   .. rubric:: Input parameters
+
+   queue
+      The SYCL command queue which will be used for SYCL kernels execution.
+
+   p_spmv_descr
+      The address of the ``p_spmv_descr`` object to be initialized. Must only be
+      called on an uninitialized ``spmv_descr_t`` object.
+
+.. container:: section
+
+   .. rubric:: Output parameters
+
+   p_spmv_descr
+      On return, the address is updated to point to a newly allocated and
+      initialized ``spmv_descr_t`` object that can be used to perform spmv.
+
+.. container:: section
+
+   .. rubric:: Throws
+
+   This function shall throw the following exceptions if the associated
+   condition is detected. An implementation may throw additional
+   implementation-specific exception(s) in case of error conditions not covered
+   here.
+
+   | :ref:`oneapi::mkl::host_bad_alloc<onemkl_exception_host_bad_alloc>`
+   | :ref:`oneapi::mkl::invalid_argument<onemkl_exception_invalid_argument>`
+   | :ref:`oneapi::mkl::unimplemented<onemkl_exception_unimplemented>`
+   | :ref:`oneapi::mkl::unsupported_device<onemkl_exception_unsupported_device>`
+
+.. _onemkl_sparse_release_spmv_descr:
+
+release_spmv_descr
+------------------
+
+.. rubric:: Syntax
+
+.. code-block:: cpp
+
+   namespace oneapi::mkl::sparse {
+
+       void release_spmv_descr (sycl::queue                       &queue,
+                                oneapi::mkl::sparse::spmv_descr_t spmv_descr);
+
+   }
+
+.. container:: section
+
+   .. rubric:: Input parameters
+
+   queue
+      The SYCL command queue which will be used for SYCL kernels execution.
+
+   spmv_descr
+      Descriptor initialized with ``init_spmv_descr``.
+
+.. container:: section
+
+   .. rubric:: Throws
+
+   This function shall throw the following exceptions if the associated
+   condition is detected. An implementation may throw additional
+   implementation-specific exception(s) in case of error conditions not covered
+   here.
+
+   | :ref:`oneapi::mkl::invalid_argument<onemkl_exception_invalid_argument>`
+   | :ref:`oneapi::mkl::unimplemented<onemkl_exception_unimplemented>`
+   | :ref:`oneapi::mkl::unsupported_device<onemkl_exception_unsupported_device>`
+
+.. _onemkl_sparse_spmv_alg:
+
+spmv_alg
+--------
+
+.. rubric:: Syntax
+
+.. code-block:: cpp
+
+   namespace oneapi::mkl::sparse {
+
+       enum class spmv_alg {
+           default_alg,
+           coo_alg1,
+           coo_alg2,
+           csr_alg1,
+           csr_alg2,
+           csr_alg3,
+       };
+
+   }
+
+.. list-table::
+   :header-rows: 1
+   :widths: 10 30 45
+
+   * - Value
+     - Description
+     - Backend equivalent
+   * - ``default_alg``
+     - Default algorithm.
+     - | MKL: N/A
+       | cuSPARSE: ``CUSPARSE_SPMV_ALG_DEFAULT``
+       | rocSPARSE: ``rocsparse_spmv_alg_default``
```

Comment:
I believe this table belongs in documentation of library implementations of the oneAPI specification, such as [`oneMKL Interfaces`](https://github.com/oneapi-src/oneMKL/), and not in the specification itself. I think that maybe the spec itself should not mention backend libraries-specific items or even imply/assume their existence as far as possible, even though the spec needs to define APIs with such backend libraries in mind. By specifying certain libraries such as Intel oneMKL/cuSPARSE/rocSPARSE (thereby leaving out others), we may inadvertently appear to endorse or be partial toward such specific libraries even though the spec is not supposed to be that way.

Here, I believe that we should just mention that "these algorithm enums are provided in case backends would like to implement various different algorithms for the operation. Behavior of the algorithms (e.g., bitwise reproducibility, atomics usage) and the preconditions to using specific algorithms (e.g. sortedness of matrix arrays) is implementation-defined and must be documented in the library implementing the oneAPI specification".

Reply:
Done in https://github.com/oneapi-src/oneAPI-spec/pull/522/commits/652bfe7592a554cd033c894074a20d391d9d7883